packages/react-components/react-message-bar/library/src/components/MessageBarGroup/MessageBarGroup.motions.tsx

```diff
@@ -0,0 +1,70 @@
+import { motionTokens, createPresenceComponent, PresenceDirection, AtomMotion } from '@fluentui/react-motion';
+
+// TODO: import these atoms from react-motion-components-preview once they're available there
+
+/**
+ * Generates a motion atom object for a fade in or fade out.
+ * @param direction - The functional direction of the motion: 'enter' or 'exit'.
+ * @param duration - The duration of the motion in milliseconds.
+ * @param easing - The easing curve for the motion. Defaults to `motionTokens.curveLinear`.
+ * @returns A motion atom object with opacity keyframes and the supplied duration and easing.
+ */
+const fadeAtom = (
+  direction: PresenceDirection,
+  duration: number,
+  easing: string = motionTokens.curveLinear,
+): AtomMotion => {
+  const keyframes = [{ opacity: 0 }, { opacity: 1 }];
+  if (direction === 'exit') {
+    keyframes.reverse();
+  }
+  return {
+    keyframes,
+    duration,
+    easing,
+  };
+};
+
+/**
+ * Generates a motion atom object for an X or Y translation, from a specified distance to zero.
+ * @param direction - The functional direction of the motion: 'enter' or 'exit'.
+ * @param axis - The axis of the translation: 'X' or 'Y'.
+ * @param distance - The distance to slide; it can be a percentage or pixels.
+ * @param duration - The duration of the motion in milliseconds.
+ * @param easing - The easing curve for the motion. Defaults to `motionTokens.curveDecelerateMid`.
+ */
+const slideAtom = (
+  direction: PresenceDirection,
+  axis: 'X' | 'Y',
+  distance: string,
+  duration: number,
+  easing: string = motionTokens.curveDecelerateMid,
+): AtomMotion => {
+  const keyframes = [{ transform: `translate${axis}(${distance})` }, { transform: 'translate${axis}(0)' }];
+  if (direction === 'exit') {
+    keyframes.reverse();
+  }
+  return {
+    keyframes,
+    duration,
+    easing,
+  };
+};
+
+export const SlideInFadeOut = createPresenceComponent(() => {
+  const duration = motionTokens.durationGentle;
+
+  return {
+    enter: [fadeAtom('enter', duration), slideAtom('enter', 'Y', '-100%', duration)],
```

Comment:
I am a big fun of functions with cryptic arguments:
- Order matters 👎 
- Error prone as it's miss-use the order 👎 
- Code does not self-describe 👎 

Compare:

```ts
slideAtom('enter', 'Y', '-100%', duration)
// vs
slideAtom({
  direction: 'enter',
  axis: 'Y',
  distance: '-100%',
  duration
})
```

Reply:
@layershifter Positional arguments are crucial for when we have a lot of optional arguments, and it would be cumbersome to fill in some just to get to the ones you need to supply.  But using named parameters for every function is not necessarily the best solution everywhere; there are trade-offs.
- Named parameters cannot be minified so nicely. Ordered parameters can minify to tiny names. This effect can cascade through call sites as well.
- These are not the dark JavaScript-only days; the code is strongly typed. 🙂 Seeing the description and order is a hover away.
- In your example, any changing of the order would not pass the compiler.
- If somehow one string parameter were swapped for another, the slide would not work at all. The mistake would be obvious. 

Of course, I can convert the atoms back to named parameters. But I find an elegance and beauty in these positional parameter call sites. It becomes a kind of DSL, whose density allows us to efficiently see how the atoms fit together. It's a bit like writing **H₂O** instead of **hydrogen-hydrogen-oxygen**. And seeing them minified so compactly is also satisfying.

```typescript
const enterAtoms: AtomMotion[] = [
  sizeEnterAtom(orientation, enterSizeDuration, enterEasing, element),
  whitespaceAtom('enter', orientation, enterSizeDuration, enterEasing),
];
```

Here, most arguments are named by the call site. So this format is less redundant than the named parameter equivalent:

```typescript
const enterAtoms: AtomMotion[] = [
  sizeEnterAtom({ orientation, duration: enterSizeDuration, easing: enterEasing, element }),
  whitespaceAtom({ direction: 'enter', orientation, duration: enterSizeDuration, easing: enterEasing }),
];
```
And enforcing the order of parameters has its own trade-offs; it's not all negative. Ordered arguments cause the call sites to read consistently; the order has a rhythm and hierarchy to it. 

**Closing question: does Fluent v9 have a rule about using named parameters instead of positional?**

Reply:
> If somehow one string argument were swapped for another, the slide would not work at all. The mistake would be obvious.

I don't think so, check #33421 🐱 

> Closing question: does Fluent v9 have a rule about using named parameters instead of positional?

No

Reply:
> > If somehow one string argument were swapped for another, the slide would not work at all. The mistake would be obvious.
> 
> I don't think so, check #33421 🐱

I don't see the correlation between the [33421 diff](https://github.com/microsoft/fluentui/pull/33421/files) and what we're discussing here (a string argument being swapped for another in a call site of ordered parameters).

Reply:
I have another [draft PR](https://github.com/microsoft/fluentui/pull/33421/files) where I explore the impact of this atom refactoring on `Collapse`:

![image](https://github.com/user-attachments/assets/7c8239e3-88a9-4e36-8068-7c33532d7f07)

Reply:
@layershifter Refactored back to named arguments in https://github.com/microsoft/fluentui/pull/33465/commits/aba26a08496b8fe853b5d47e2bfab7a87cd01407 .